File: be_fsharp_dotnet_core/Startup.fs
namespace TodoList

open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Cors.Infrastructure
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting

type Startup private () =

    let corsConfiguration (builder : CorsPolicyBuilder) =
        builder.WithOrigins("http://localhost:3000")
            .AllowAnyMethod()
            .AllowAnyHeader()
            |> ignore

    new (configuration: IConfiguration) as this =
        Startup() then
        this.Configuration <- configuration

    // This method gets called by the runtime. Use this method to add services to the container.
    member this.ConfigureServices(services: IServiceCollection) =
        // Add framework services.
        services.AddControllers() |> ignore
        services.AddCors() |> ignore
        services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_3_0) |> ignore

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    member this.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
        if (env.IsDevelopment()) then
            app.UseDeveloperExceptionPage() |> ignore

        app.UseCors(corsConfiguration) |> ignore
        app.UseHttpsRedirection() |> ignore
        app.UseRouting() |> ignore

        app.UseAuthorization() |> ignore

        app.UseEndpoints(fun endpoints ->
            endpoints.MapControllers() |> ignore
            ) |> ignore

    member val Configuration : IConfiguration = null with get, set

File: be_csharp_dotnet/Startup.cs
using System;
using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using TodoList.Controllers;

namespace TodoList
{
    public class Startup
    {
        private const String TodoListOrigins = "_TodolistOrigins";
        private readonly IConfiguration _configuration;

        public Startup( IConfiguration configuration ) =>
            _configuration = configuration;

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure( IApplicationBuilder app, IHostEnvironment env )
        {
            if ( env.IsDevelopment() )
                app.UseDeveloperExceptionPage();

            app.UseCors( TodoListOrigins );
            app.UseHttpsRedirection();
            app.UseRouting();
            app.UseEndpoints( endpoints => endpoints.MapControllers() );
        }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices( IServiceCollection services )
        {
            services
                .AddCors( options => options
                              .AddPolicy( TodoListOrigins,
                                          builder => builder
                                                     .WithOrigins( "http://localhost:3000" )
                                                     .AllowAnyHeader()
                                                     .AllowAnyMethod() ) );
            services.AddControllers();

            services.AddSingleton<IRepo, Repo>();
            //services.AddTransient<IRepo, Repo>();
        }
    }
}